control/mesh.proto

```diff
@@ -183,3 +184,264 @@ message ScanNetworksRequest {
 message ScanNetworksReply {
   repeated NetworkInfo networks = 1;
 }
+
+// Diagnostics
+
+enum DiagnosticType {
+  option allow_alias = true;
+  MAC_EXTENDED_ADDRESS = 0;
+  RLOC = 1;
+  MAC_ADDRESS = 1;
+  MODE = 2;
+  TIMEOUT = 3;
+  CONNECTIVITY = 4;
+  ROUTE64 = 5;
+  LEADER_DATA = 6;
+  NETWORK_DATA = 7;
+  IPV6_ADDRESS_LIST = 8;
+  MAC_COUNTERS = 9;
+  BATTERY_LEVEL = 14;
+  SUPPLY_VOLTAGE = 15;
+  CHILD_TABLE = 16;
+  CHANNEL_PAGES = 17;
+  TYPE_LIST = 18;
+  MAX_CHILD_TIMEOUT = 19;
```

Comment:
Let's add a comment that the values of this enum should match the values of the OT's `NetworkDiagnosticTlv::Type` enum